tests/moduletool/test_update.py

```diff
@@ -66,3 +77,87 @@ def test_module_update_with_install_mode_master(
     assert os.path.exists(extra_file_mod2) == mod2_should_be_updated
     extra_file_mod8 = os.path.join(libs_folder, "mod8", file_name_extra_file)
     assert os.path.exists(extra_file_mod8) == mod8_should_be_updated
+
+
+@pytest.mark.parametrize("corrupt_module", [False, True])
+def test_module_update_with_v2_module(
+    tmpdir: py.path.local, modules_v2_dir: str, snippetcompiler_clean, modules_repo: str, corrupt_module: bool
+) -> None:
+    """
+    Assert that the `inmanta module update` command works correctly when executed on a project with a V2 module.
+
+    :param corrupt_module: Whether the module to be updated contains a syntax error or not.
+    """
+    template_module_name = "elaboratev2module"
+    template_module_dir = os.path.join(modules_v2_dir, template_module_name)
+
+    pip_index = PipIndex(artifact_dir=os.path.join(str(tmpdir), "pip-index"))
+
+    def assert_version_installed(module_name: str, version: str) -> None:
+        package_name = ModuleV2.get_package_name_for(module_name)
+        installed_packages: Dict[str, Version] = process_env.get_installed_packages()
+        assert package_name in installed_packages
+        assert str(installed_packages[package_name]) == version
+
+    for module_name, versions in {"module2": ["2.0.1", "2.1.0", "2.2.0", "3.0.1"], "module1": ["1.2.4", "1.2.5"]}.items():
```

Comment:
This is really cool!

I would document why the module structure is set up like this specifically, because I have the feeling that some subtly eludes me.